src/ipc/context.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_IPC_CONTEXT_H
+#define BITCOIN_IPC_CONTEXT_H
+
+namespace ipc {
+//! Context struct used to give IPC protocol implementations or implementation
+//! hooks access to application state, in case they need to run extra code that
+//! that isn't needed within a single procoess, like code copying global state
+//! from an existing process to a new process when it's initialized, or code
+//! dealing with shared objects that are created or destroyed remotely.
+struct Context
```

Comment:
> I am using the word "context" in the same way in wallet, node, and IPC code. I think "context" is different than "state" because can "state" refer can refer to any type of data including temporary data, while "context" refers to longer lasting environmental data. Like data from a high-level outer environment that is is passed through low level code, but the low level code just treats as opaque. So "context" seems more appropriate than "state". And LowLevelContext seems incorrect because the context struct is holding higher-level bitcoin-specific data that the lower-level IPC code is treats as opaque. It is high level data, not low level data.

Thanks for the explanation, I agree it's servicing the same purpose than `NodeContext`/`WalletContext`, namely avoiding repeated variables/parameters declarations and proliferating usage of globals and "context" is a good pick. Prefixing would be still be better imho, like `ApplicationContext`/`ServerContext` to hint the lack of abstract/subclass relationships between current `Context` and `NodeContext`.